source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -126,7 +142,10 @@ static const QuicLogLevel DFATAL = FATAL;
 
 class QuicLogEmitter {
 public:
-  explicit QuicLogEmitter(QuicLogLevel level);
+  // |file_name| and |function_name| MUST be valid for the lifetime of the QuicLogEmitter. This is
+  // guaranteed when passing __FILE__ and __func__.
+  explicit QuicLogEmitter(QuicLogLevel level, const char* file_name, int line,
```

Comment:
`explicit` is not required when there are multiple arguments, but it's still good practice because it prevents initializer-list construction. The Google style guide explicitly says it's OK either way in this case, and the Envoy one doesn't mention it. Is that OK with you?

`__FILE__` and `__func__` are expanded by the preprocessor from the caller of the macro that lead to them. We can't hard-code `__FILE__` and `__func__` in the .cc file because that'll give you that file name and function, as opposed to this macro usage which gives us the actual caller.

Reply:
I see. Thanks for explanation!